pkg/account/api/search_filter_test.go

```diff
@@ -376,6 +389,64 @@ func TestCreateSearchFilter(t *testing.T) {
 			},
 			expectedErr: nil,
 		},
+		{
+			desc: "success: system admin account",
+			setup: func(s *AccountService) {
+				email := "bucketeer@example.com"
+				orgID := "system_admin_org_id"
+				acc := &domain.AccountV2{
+					AccountV2: &accountproto.AccountV2{
+						Email:            email,
+						OrganizationId:   orgID,
+						OrganizationRole: accountproto.AccountV2_Role_Organization_MEMBER,
+						EnvironmentRoles: []*accountproto.AccountV2_EnvironmentRole{
+							{
+								EnvironmentId: "envID0",
+								Role:          accountproto.AccountV2_Role_Environment_VIEWER,
+							},
+						},
+					},
+				}
+				s.accountStorage.(*accstoragemock.MockAccountStorage).EXPECT().GetAccountV2ByEnvironmentID(
+					gomock.Any(), "email", gomock.Any(),
+				).Return(acc, nil)
+
+				s.accountStorage.(*accstoragemock.MockAccountStorage).EXPECT().GetSystemAdminAccountV2(
+					gomock.Any(), email,
+				).Return(acc, nil)
+
+				s.accountStorage.(*accstoragemock.MockAccountStorage).EXPECT().GetAccountV2(
+					gomock.Any(), email, orgID,
```

Comment:
Here I ensure that the organization ID is the system admin organization and not the ID from the request.